src/fmu/dataio/dataio.py

```diff
@@ -728,6 +730,16 @@ def _update_fmt_flag(self) -> None:
 
     def _update_check_settings(self, newsettings: dict) -> None:
         """Update instance settings (properties) from other routines."""
+        # if no newsettings (kwargs) this rutine is not needed
+        if not newsettings:
+            return
+
+        warnings.warn(
```

Comment:
this warning was just moved since we need to run this `_update_check_settings` several places. The entire function can be removed once we move towards fmu.dataio 3.0